### PR TITLE
Dornerworks x64 VM

### PR DESCRIPTION
**Pull request by nomadeel**
This PR contains commits from Dornerworks' x64 VM patches for camkes-vm-linux: https://github.com/dornerworks/camkes-vm-linux/tree/open_source_release. This PR also contains a commit which includes a binary image of the 64-bit Linux kernel built using the config as found in the PR.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -48,11 +55,11 @@ if(CAmkESVMDefaultBuildrootOverlay)
     # Linux config and symvers are to be copied to unpacked archive
     set(
         linux_config
-        "${CMAKE_CURRENT_SOURCE_DIR}/linux_configs/${linux_major}.${linux_minor}/config"
+        "${CMAKE_CURRENT_SOURCE_DIR}/linux_configs/${linux_major}.${linux_minor}/${arch_size_dir}/config"
     )
     set(
         linux_symvers
-        "${CMAKE_CURRENT_SOURCE_DIR}/linux_configs/${linux_major}.${linux_minor}/Module.symvers"
+        "${CMAKE_CURRENT_SOURCE_DIR}/linux_configs/${linux_major}.${linux_minor}/${arch_size_dir}/Module.symvers"
```
Comment (oliver-wm) [inline CMakeLists.txt:62]: Is arch_size_dir == 64 || 32?

Comment (oliver-wm) [reply CMakeLists.txt:62]: AH, didn't see above code

Comment (nomadeel) [reply CMakeLists.txt:62]: Yeah, you beat me to it

File: vm-linux-helpers.cmake [hunk 2]
```diff
@@ -227,15 +227,26 @@ endfunction(UpdateDtbFromInitrd)
...
 endfunction(GetDefaultLinuxKernelFile)
 
 # Function for getting the default location of the Linux guest rootfs provided
 # by the vm project
 # dest_file: caller variable which is set with the rootfs location
 function(GetDefaultLinuxRootfsFile dest_file)
+  if(KernelSel4ArchX86_64)
     set(
-        ${dest_file} ${VM_LINUX_PROJECT_DIR}/images/rootfs/default_buildroot_rootfs.cpio
+        ${dest_file} ${VM_LINUX_PROJECT_DIR}/images/rootfs/64/default_buildroot_rootfs-bare.cpio
```
Comment (oliver-wm) [inline vm-linux-helpers.cmake:243]: can you use something like `arch_size_dir` here to remove the conditional?

File: vm-linux-helpers.cmake [hunk 3]
```diff
@@ -227,15 +227,26 @@ endfunction(UpdateDtbFromInitrd)
 # by the vm project
 # dest_file: caller variable which is set with the kernel location
 function(GetDefaultLinuxKernelFile dest_file)
-    set(${dest_file} ${VM_LINUX_PROJECT_DIR}/images/kernel/default_bzimage_4.8.16 PARENT_SCOPE)
+  if(KernelSel4ArchX86_64)
+    set(${dest_file} ${VM_LINUX_PROJECT_DIR}/images/kernel/64/default_bzimage_4.8.16 PARENT_SCOPE)
```
Comment (oliver-wm) [inline vm-linux-helpers.cmake:231]: and here

**Review by oliver-wm — APPROVED** [review]
Looks good to me

**Comment by nomadeel** [comment]
Uhhhhh, I'm not sure how I closed this and didn't merge it but I'll re-open it.

**Comment by lsf37** [comment]
If you have rebase it and it closed because of that, it usually means that it was already merged on bitbucket.

**Comment by nomadeel** [comment]
It's not on Bitbucket either, so I probably accidentally closed it.

**Comment by lsf37** [comment]
I don't think so, the PR shows 0 commits. That usually means they are already merged. Let me check if I can find it in the history.

**Comment by lsf37** [comment]
Unless, of course, you set your branch to be equal to master and thereby made it 0 commits

**Comment by nomadeel** [comment]
Seems to be it: https://github.com/nomadeel/camkes-vm-linux/commits/dornerworks_x64

**Comment by nomadeel** [comment]
Yeah, somehow I force-pushed master to my fork and made it so the head was a few commits back from master.

Ah well, re-opening this now, also added a commit to update the 32-bit image to have driver support for VirtIO block devices.

File: vm-linux-helpers.cmake (hunk 2 above)
Comment (nomadeel) [reply vm-linux-helpers.cmake]: Done.

**Pull request merged**